src/Environment.php

```diff
@@ -490,31 +490,21 @@ public function isTemplateFresh($name, $time)
      *
      * @param string|TemplateWrapper|array $names A template or an array of templates to try consecutively
      *
-     * @return TemplateWrapper|Template
+     * @return TemplateWrapper
      *
      * @throws LoaderError When none of the templates can be found
      * @throws SyntaxError When an error occurred during compilation
      */
     public function resolveTemplate($names)
     {
         if (!\is_array($names)) {
-            $names = [$names];
+            return $this->load($names);
         }
 
         foreach ($names as $name) {
-            if ($name instanceof Template) {
-                return $name;
-            }
-            if ($name instanceof TemplateWrapper) {
-                return $name;
-            }
-
             try {
-                return $this->loadTemplate($name);
+                return $this->load($name);
             } catch (LoaderError $e) {
-                if (1 === \count($names)) {
```

Comment:
is it expected to remove this direct re-throwing in case of resolving a single template name ? This would hurt DX by replacing the original message with `Unable to find one of the following templates: "foo.html.twig".`

Reply:
We don't remove the throw as in the case of 1 template, it's thrown in the `if` before the `foreach`.

Reply:
hmm, indeed, I missed that.